Sources/SwiftMemcache/Extensions/ByteBuffer+SwiftMemcache.swift

```diff
@@ -25,3 +25,60 @@ extension ByteBuffer {
         self.writeString(string)
     }
 }
+
+extension ByteBuffer {
+    /// Reads an integer from ASCII characters directly from this `ByteBuffer`.
+    /// The reading stops as soon as a non-digit character is encountered.
+    ///
+    /// - Returns: A `UInt64` integer read from the buffer.
+    /// If the buffer does not contain any digits at the current reading position, returns `nil`.
+    mutating func readIntegerFromASCII() -> UInt64? {
+        var value: UInt64 = 0
+        while self.readableBytes > 0, let currentByte = self.readInteger(as: UInt8.self),
+              currentByte >= UInt8.zero && currentByte <= UInt8.nine {
+            value = (value * 10) + UInt64(currentByte - UInt8.zero)
+        }
+        return value > 0 ? value : nil
+    }
+}
+
+extension ByteBuffer {
+    /// Serialize and writes MemcachedFlags to the ByteBuffer.
+    ///
+    /// This method runs a loop over the flags contained in a MemcachedFlags instance.
+    /// For each flag that is set to true, its corresponding byte value,
+    /// followed by a whitespace character, is added to the ByteBuffer.
+    ///
+    /// - parameters:
+    ///     - flags: An instance of MemcachedFlags that holds the flags intended to be serialized and written to the ByteBuffer.
+    mutating func writeMemcachedFlags(flags: MemcachedFlags) {
+        if let shouldReturnValue = flags.shouldReturnValue, shouldReturnValue {
+            self.writeInteger(UInt8.whitespace)
+            self.writeInteger(UInt8.v)
+        }
+    }
+}
+
+extension ByteBuffer {
+    /// Parses flags from this `ByteBuffer`, advancing the reader index accordingly.
+    ///
+    /// - returns: A `MemcachedFlags` instance populated with the flags read from the buffer.
+    mutating func readMemcachedFlags() -> MemcachedFlags {
+        var flags = MemcachedFlags()
+        while let nextByte = self.readInteger(as: UInt8.self) {
+            switch nextByte {
+            case UInt8.v:
+                flags.shouldReturnValue = true
+            case UInt8.whitespace:
+                continue
+            case UInt8.carriageReturn:
+                guard let _ = self.readInteger(as: UInt8.self), self.readableBytes > 0 else {
+                    break
+                }
+            default:
+                preconditionFailure("Unrecognized flag.")
+            }
+        }
+        return flags
+    }
```

Comment:
I just added a comment further down that basically says we should try to not do 2 yet since it makes everything a bit more complicated. However can you please add a test where you feed a partial buffer in that doesn't have everything i.e. only a `\r` and not the `\n` afterwards and then pass it in after and assert the right things happen.

Reply:
This will be handled in a future PR!